src/protagonist/API/Features/Storage/StorageController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using API.Features.Storage.Converters;
+using API.Features.Storage.Requests;
+using API.Infrastructure;
+using API.Settings;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.Storage;
+
+/// <summary>
+/// DLCS REST API Operations for customer storage objects
+/// </summary>
+public class StorageController : HydraController
+{
+    public StorageController(
+        IMediator mediator,
+        IOptions<ApiSettings> settings) : base(settings.Value, mediator)
+    {
+    }
+
+    /// <summary>
+    /// Gets the storage object of an image within a customer's space
+    /// </summary>
+    [HttpGet]
+    [Route("/customers/{customerId}/spaces/{spaceId}/images/{imageId}/storage")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetImageStorage(
+        [FromRoute] int customerId,
+        [FromRoute] int spaceId,
+        [FromRoute] string imageId,
+        CancellationToken cancellationToken)
+    {
+        var request = new GetImageStorage(customerId, spaceId, imageId);
+        
+        return await HandleFetch(request, 
+            s => s.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to get storage from image",
```

Comment:
Minor thing - this is the storage _for_ an image, rather than _from_ an image (same with space, below).